src/components/app/data/services/contentHighlights.js

```diff
@@ -41,18 +39,12 @@ export async function fetchContentHighlights(enterpriseUUID, options = {}) {
     enterprise_customer: enterpriseUUID,
     ...options,
   });
-
   const url = `${getConfig().ENTERPRISE_CATALOG_API_BASE_URL}/api/v1/highlight-sets/?${queryParams.toString()}`;
-
   try {
-    const response = await getAuthenticatedHttpClient().get(url);
-    const data = camelCaseObject(response.data);
-    return data.results ?? null;
+    const { results } = await fetchPaginatedData(url);
+    return results;
   } catch (error) {
-    const errorResponseStatusCode = getErrorResponseStatusCode(error);
-    if (errorResponseStatusCode === 404) {
```

Comment:
[inform] API does not return a 404 when there are no results; no need to check 404 status code explicitly.